compiler/semtypes.nim

```diff
@@ -555,11 +555,15 @@ proc semIdentWithPragma(c: PContext, kind: TSymKind, n: PNode,
     else: discard
   else:
     result = semIdentVis(c, kind, n, allowed)
+    let invalidPragmasForPush = if fromTopLevel and "`gensym" notin result.name.s:
```

Comment:
Could we maybe remove the check for `` `gensym `` here and mangle the output of `{.exportc.}` when no string is given to it instead? Because this errors too despite not being `skVar/skLet/skConst`:

```nim
{.push exportc.}

template foo =
  proc bar {.gensym.} = discard

foo()
```

Reply:
Too hacky.

Reply:
> Too hacky.

How should we check `gensym`? There is not a flag dedicated for it since it is a newly created ident. 

```nim
if sfGenSym in s.flags:
  result.add newIdentNode(getIdent(c.ic, x.name.s & "`gensym" & $c.instID),
    if c.instLines: actual.info else: templ.info)
```

>  mangle the output of {.exportc.} when no string is given to it instead

I would like to revive https://github.com/nim-lang/Nim/pull/19636 with some adaptations

Reply:
Introduce a new flag like `sfWasGenSym`.

Reply:
Okay, I will move `` `gensym`` check into `newSymG` and incl the flag there.

Reply:
At least make the check cheaper by doing:

```nim

getIdent(c.ic, "`" & x.name.s & "`gensym" & $c.instID)
...

s.name.s[0] == '`'

```